src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -247,20 +247,40 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
             resolver.resolve_str_path_error(DUMMY_SP, &path_str, ns, module_id)
         });
         debug!("{} resolved to {:?} in namespace {:?}", path_str, result, ns);
-        let result = match result {
-            Ok((_, Res::Err)) => Err(()),
-            x => x,
+        let result = match result.map(|(_, res)| res) {
+            Ok(Res::Err) | Err(()) => {
+                // resolver doesn't know about true and false so we'll have to resolve them
+                // manually as bool
+                if let Some((_, res)) = is_bool_value(path_str, ns) { Ok(res) } else { Err(()) }
+            }
+            Ok(res) => Ok(res.map_id(|_| panic!("unexpected node_id"))),
         };
 
-        if let Ok((_, res)) = result {
-            let res = res.map_id(|_| panic!("unexpected node_id"));
-            // In case this is a trait item, skip the
-            // early return and try looking for the trait.
-            let value = match res {
-                Res::Def(DefKind::AssocFn | DefKind::AssocConst, _) => true,
-                Res::Def(DefKind::AssocTy, _) => false,
+        if let Ok(res) = result {
+            match res {
+                Res::Def(DefKind::AssocFn | DefKind::AssocConst, _) => {
+                    if ns != ValueNS {
+                        return Err(ResolutionFailure::WrongNamespace(res, ns).into());
+                    } else {
+                        // In case this is a trait item, skip the
+                        // early return and try looking for the trait.
```

Comment:
Can you leave the comment but remove the `else` branch?
```suggestion
                    // In case this is a trait item, skip the
                    // early return and try looking for the trait.
```

Reply:
I kinda find it suspicious that way - there will be a fallthrough comment, but in a place where it feels a bit like it was forgotten.

Reply:
It doesn't seem forgotten to me, though? This is an explicit check, there's nothing to do if the check fails other than keep going.

Reply:
Let me see what rustfmt does with it, but I kinda remember glossing over it and thinking that `if` was an if guard and I expected all match arms to always return.